plugin/src/main/kotlin/tanvd/grazi/language/Lang.kt

```diff
@@ -1,47 +1,50 @@
 import tanvd.grazi.GraziBundle
+import tanvd.grazi.GraziPlugin
 
 @Suppress("unused")
-enum class Lang(val jLanguage: Language,
+enum class Lang(val displayName: String, val shortCode: String, val className: String,
                 private val enabledRules: Set<String> = emptySet(),
                 private val disabledRules: Set<String> = emptySet()) {
-    BRITISH_ENGLISH(BritishEnglish(), GraziBundle.langConfig("en.enabled"), GraziBundle.langConfig("en.disabled")),
-    AMERICAN_ENGLISH(AmericanEnglish(), GraziBundle.langConfig("en.enabled"), GraziBundle.langConfig("en.disabled")),
-    CANADIAN_ENGLISH(CanadianEnglish(), GraziBundle.langConfig("en.enabled"), GraziBundle.langConfig("en.disabled")),
-    RUSSIAN(Russian(), GraziBundle.langConfig("ru.enabled")),
-    PERSIAN(Persian()),
-    FRENCH(French()),
-    GERMANY_GERMAN(GermanyGerman()),
-    AUSTRIAN_GERMAN(AustrianGerman()),
-    POLISH(Polish()),
-    ITALIAN(Italian()),
-    DUTCH(Dutch()),
-    PORTUGAL_PORTUGUESE(PortugalPortuguese()),
-    BRAZILIAN_PORTUGUESE(BrazilianPortuguese()),
-    CHINESE(Chinese()),
-    GREEK(Greek()),
-    JAPANESE(Japanese()),
-    ROMANIAN(Romanian()),
-    SLOVAK(Slovak()),
-    SPANISH(Spanish()),
-    UKRAINIAN(Ukrainian());
+    BRITISH_ENGLISH("English (GB)", "en", "BritishEnglish",GraziBundle.langConfig("en.enabled"), GraziBundle.langConfig("en.disabled")),
+    AMERICAN_ENGLISH("English (US)", "en", "AmericanEnglish", GraziBundle.langConfig("en.enabled"), GraziBundle.langConfig("en.disabled")),
+    CANADIAN_ENGLISH("English (Canadian)", "en", "CanadianEnglish", GraziBundle.langConfig("en.enabled"), GraziBundle.langConfig("en.disabled")),
+    RUSSIAN("Russian", "ru", "Russian", GraziBundle.langConfig("ru.enabled")),
+    PERSIAN("Persian", "fa", "Persian"),
+    FRENCH("French", "fr", "French"),
+    GERMANY_GERMAN("German (Germany)", "de", "GermanyGerman"),
+    AUSTRIAN_GERMAN("German (Austria)", "de", "AustrianGerman"),
+    POLISH("Polish", "pl", "Polish"),
+    ITALIAN("Italian", "it", "Italian"),
+    DUTCH("Dutch", "nl", "Dutch"),
+    PORTUGAL_PORTUGUESE("Portuguese (Portugal)", "pt", "PortugalPortuguese"),
+    BRAZILIAN_PORTUGUESE("Portuguese (Brazil)", "pt", "BrazilianPortuguese"),
+    CHINESE("Chinese", "zh", "Chinese"),
+    GREEK("Greek", "el", "Greek"),
+    JAPANESE("Japanese", "ja", "Japanese"),
+    ROMANIAN("Romanian", "ro", "Romanian"),
+    SLOVAK("Slovak", "sk", "Slovak"),
+    SPANISH("Spanish", "es", "Spanish"),
+    UKRAINIAN("Ukrainian", "uk", "Ukrainian");
 
     companion object {
-        operator fun get(lang: Language): Lang? = values().find { it.shortCode == lang.shortCode }
+        operator fun get(lang: Language): Lang? = values().find { lang.name == it.displayName }
         operator fun get(code: String): Lang? = values().find { it.shortCode == code }
 
-        val sortedValues = values().sortedBy { it.displayName }
+        fun sortedValues() = values().sortedBy(Lang::displayName)
```

Comment:
`val sortedValues: List<Lang> get() = values().sortedBy...`

Reply:
Why? Now we have both functions - values and sortdeValues, but with val we have function and field - it's strange.

Reply:
Hm, well, ok :)